tools/src/main/scala/scala/scalanative/build/LLVM.scala

```diff
@@ -120,26 +120,29 @@ private[scalanative] object LLVM {
       }
     }
 
+    val (includePaths, excludePaths) = paths.partition(include(_))
+
     // delete .o files for all excluded source files
-    paths.foreach { path =>
-      if (!include(path)) {
-        val opath = Paths.get(path + oExt)
-        if (Files.exists(opath)) {
-          Files.delete(opath)
-        }
-      }
+    excludePaths.foreach { path =>
+      val opath = Paths.get(path + oExt)
+      if (Files.exists(opath)) Files.delete(opath)
     }
 
+    val fltoOpt   = flto(config)
+    val targetOpt = target(config)
+
     // generate .o files for all included source files in parallel
-    paths.par.foreach { path =>
-      val opath = path + oExt
-      if (include(path) && !Files.exists(Paths.get(opath))) {
+    includePaths.par.map { path =>
+      val opath   = path + oExt
+      val objPath = Paths.get(opath)
+      if (!Files.exists(objPath)) {
```

Comment:
Keep in mind that when you change the target triple these files need to be invalidated.
This if there will be a target triple setting changeable by the users.

Reply:
As you can see I avoided different output directories but I thought it might be best to see how this goes. I think changing the target triple in the build and `reload`ing the build probably means you should `clean`.

The build currently removes the `.ll` on every build but I am not sure of that history. It is probably just because we have no timestamp checking between `.nir` and `.ll` and it is not a one to one mapping. https://github.com/scala-native/scala-native/pull/2033/files#diff-ba1b9a87401a482acad7eb0588b7cc93ffeabdd871c5abd747b7bd5a79cf6a4fR67-R68

Now that I think about it we may need a `Discover.targetTriple` so we can have an environment variable because other wise how could you use this in a CI build? Another option might be to have `Seq(targetTriple1, targetTriple2, ...)`. I hate to over engineer a feature on its first try especially since we need experience and feedback to find the sweet spot.

Reply:
@ekrich and somewhere here we jump back to our discussion :)

I feel that we should force user to use different sbt task for different targetTriple.

Reply:
I think the currently implemented solution is good: when changing the target triple, like any other config option, the user should `clean`. There is no difference between the target triple and, say, the GC configuration.

> Now that I think about it we may need a Discover.targetTriple so we can have an environment variable because other wise how could you use this in a CI build?

Using
```
$ sbt 'set nativeConfig ~= { _.withTargetTriple("whatever") }' test
```
There's no need for an environment variable.

Reply:
That is good.

Reply:
A clean is required after changing settings at least the way things are now. A workaround is perhaps to have different projects for different targets or in CI I don't think that would be a problem because each build would be mostly clean.

Reply:
Personally I think this is a pity. One of the selling points of Scala Native compared to say GraalVM Native Image is that it could allow fast cross-compilation to generate programs for various operative systems. We're not yet there, this is sure, but I think that imposing a clean to change target is a waste of build time that will lower this benefit.
Compilation to NIR doesn't change between targets and this way if you want to cross compile you're asking to re-do this work over and over.
I hope that in the future \(when we will see usages of cross-compiling, for example [throw Zig](https://ziglang.org/#Zig-is-also-a-C-compiler)\) this decision will be revised.

Reply:
I agree that this is not ideal but we don't know when build settings change. Removing that check would slow down the normal use case. This check was there so that is not changing and I am not sure removing it is a great idea.

I am open to a `Seq["target triple"]` and individual build directories and lots of other stuff but I think it is better to have a nice change like this first and then when we prove things out we might find out that there are even better ways to do things.

Reply:
If we do anything about this, it should be the same way that we invalidate the output of the linker if the config has changed in Scala.js:
https://github.com/scala-js/scala-js/blob/a27c22bc0bf15a84b14745231815ea0a4a12c90f/sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPluginInternal.scala#L188-L205

But that is for a different PR.

Reply:
Thank you @ekrich and @sjrd .
As you said, Eric, this is a good starting point to start playing with cross-compilation and it doesn't necessarily mean that can't be improved in the future when if we find better ways to do things.